Dockerfile

```diff
@@ -27,6 +27,19 @@ COPY todo_app /srv/todoapp/todo_app
 WORKDIR /srv/todoapp
 RUN scripts/build.sh
 
+FROM poetry as test 
+RUN apt-get update
+RUN apt-get -y install firefox-esr
+RUN wget https://github.com/mozilla/geckodriver/releases/download/v0.31.0/geckodriver-v0.31.0-linux32.tar.gz
+RUN tar -xzf geckodriver-v0.31.0-linux32.tar.gz
+RUN mv geckodriver /bin/geckodriver
+COPY pyproject.toml pyproject.toml
+COPY poetry.lock poetry.lock
```

Comment:
Note that you can do this in a single layer: `COPY pyproject.toml poetry.lock ./`